### ISSUE TITLE
[runtime] accept any type of tensor when running CompiledModel

### ISSUE THREAD

**Issue by pilkicTT** [tenstorrent/tt-forge-fe#161]
Currently, we expect input tensors to be provided as torch tensors. Which is not ideal, when user is using tensorflow or forge modules.

**Comment by staylorTT** [comment]
@nvukobratTT I see the draft PR seems abandoned, did this get fixed?

**Comment by nvukobratTT** [comment]
Yes, this is the part that I want the core compiler team to define before we push it further.

**Comment by skhatriTT** [comment]
@nvukobratTT any progress on this one?
